connector/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/ConsumerStrategy.scala

```diff
@@ -127,7 +126,7 @@ private[kafka010] case class SubscribePatternStrategy(topicPattern: String)
       kafkaParams: ju.Map[String, Object]): Consumer[Array[Byte], Array[Byte]] = {
     val updatedKafkaParams = setAuthenticationConfigIfNeeded(kafkaParams)
     val consumer = new KafkaConsumer[Array[Byte], Array[Byte]](updatedKafkaParams)
-    consumer.subscribe(ju.regex.Pattern.compile(topicPattern), new NoOpConsumerRebalanceListener())
```

Comment:
After https://github.com/apache/kafka/pull/14638/files
the class `NoOpConsumerRebalanceListener` has been removed

https://github.com/apache/kafka/blob/bdd85405e3d72b5089d99026ca1e5aa5b3bc3e51/clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java#L695-L696
<img width="948" alt="image" src="https://github.com/apache/spark/assets/15246973/3bce12b2-e9fd-4f05-8cc6-5d94b0737b05">